eureka-core/src/main/java/com/netflix/eureka/EurekaBootStrap.java

```diff
@@ -222,7 +223,11 @@ protected void initEurekaServerContext() throws Exception {
         EurekaMonitors.registerAllStats();
     }
 
-    protected abstract EurekaHttpClient getEurekaHttpClient();
+    protected AbstractDiscoveryClientOptionalArgs<?> getDiscoveryClientOptionalArgs() {
```

Comment:
We should probably have a discussion about the name of this class `AbstractDiscoveryClientOptionalArgs` (or a refactor) as `TransportClientFactories` is no longer optional.